flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreExpireImpl.java

```diff
@@ -91,7 +110,7 @@ public void expire() {
                                 <= millisRetained) {
                     // within time threshold, can assume that all snapshots after it are also within
                     // the threshold
-                    expireUntil(id);
+                    expireUntil(earliest, id);
```

Comment:
Move the two calls for `writeEarliestHint` to here. `id` will be the earliest snapshot remaining no matter how.

Reply:
For above handle for "No expire happens", leave in `expireUntil`